zerver/migrations/0530_sessions_data_migration.py

```diff
@@ -0,0 +1,108 @@
+        return
+
+    session_key_index_name = ""
+    with transaction.atomic():
+        with connection.cursor() as cursor:
+            # Fetch indexes names.
+            cursor.execute(
+                SQL("""SELECT indexname FROM pg_indexes WHERE tablename = 'zerver_realmsession' """)
+            )
+
+            indexes_tuples = cursor.fetchall()
+            # Drop all indexes.
+            for index in indexes_tuples:
+                if index[0] == "zerver_realmsession_pkey":
+                    query: Composable = SQL(
+                        """ALTER TABLE zerver_realmsession drop constraint zerver_realmsession_pkey"""
+                    )
+                else:
+                    if "session_key" in index[0]:
+                        session_key_index_name = index[0]
+
+                    query = SQL("""DROP INDEX {index_name}""").format(
+                        index_name=Identifier(index[0])
+                    )
+
+                cursor.execute(query)
+
+            # Copy over data.
+            cursor.execute(SQL("INSERT INTO zerver_realmsession (SELECT * FROM django_session)"))
+
+            # If Insertion is successful, Should we clear django_session table ?
+            # cursor.execute(SQL("TRUNCATE django_session)"))
+
+    # Create back the indexes
+    with transaction.atomic():
+        with connection.cursor() as cursor:
+            # Primary key creates an index by default.
+            cursor.execute(
+                SQL(
+                    """
+                    ALTER TABLE zerver_realmsession ADD CONSTRAINT
+                        zerver_realmsession_pkey PRIMARY KEY (session_key)
+                    """
+                )
+            )
+
+            cursor.execute(
+                SQL(
+                    # {index_name} was auto generated by django, we create the index with the same name just in case,
+                    # as i'm not sure if a custom name in this case would be a problem.
+                    """
+                    CREATE INDEX IF NOT EXISTS {index_name}
+                        ON zerver_realmsession (session_key varchar_pattern_ops);
+
+                    CREATE INDEX IF NOT EXISTS zerver_realmsession_expire_date_idx
+                        ON zerver_realmsession (expire_date);
+
+                    CREATE INDEX IF NOT EXISTS zerver_realmsession_realm_idx
+                        ON zerver_realmsession (realm_id);
+
+                    CREATE INDEX IF NOT EXISTS zerver_realmsession_user_idx
+                        ON zerver_realmsession (user_id);
+
+                    CREATE INDEX IF NOT EXISTS zerver_realmsession_realm_user_idx
+                        ON zerver_realmsession (realm_id, user_id);
+
+                    CREATE INDEX IF NOT EXISTS zerver_realmsession_ip_address_idx
+                        ON zerver_realmsession (ip_address);
+                    """
+                ).format(index_name=Identifier(session_key_index_name))
```

Comment:
We have the precise index definitions that we dropped, in the `indexdef` column:
```
zulip=> select * from pg_indexes where tablename = 'django_session';
 schemaname |   tablename    |                indexname                 | tablespace |                                                          indexdef
------------+----------------+------------------------------------------+------------+-----------------------------------------------------------------------------------------------------------------------------
 zulip      | django_session | django_session_session_key_c0390e0f_like |            | CREATE INDEX django_session_session_key_c0390e0f_like ON zulip.django_session USING btree (session_key varchar_pattern_ops)
 zulip      | django_session | django_session_pkey                      |            | CREATE UNIQUE INDEX django_session_pkey ON zulip.django_session USING btree (session_key)
 zulip      | django_session | django_session_expire_date_a5c62663      |            | CREATE INDEX django_session_expire_date_a5c62663 ON zulip.django_session USING btree (expire_date)
(3 rows)
```
I'd much rather not hard-code these, because it's very easy to accidentally have had drift.

Dropping the primary key constraint makes me leery -- I think my proposal is that we drop everything else, storing the `indexdef` statements as we go, and then re-run just those statements at the end.

Reply:
I agree that we should avoid hard coding and go with executing those `indexdef`s instead.

But why are you leery of dropping the primary key ?

Reply:
But why are you leery of dropping the primary key ?